### PR TITLE
Add tests to pkg/commands/git - Part 1

### PR DESCRIPTION
**Pull request by antham**

### CODE REVIEW

**Comment by antham** [comment]
This file is quite big so let's break it in sub PR's, it's going to be easier, if you can review this @jesseduffield  pls

File: pkg/commands/git.go [hunk 1]
```diff
@@ -121,28 +121,28 @@ func (c *GitCommand) MergeStatusFiles(oldFiles, newFiles []File) []File {
 		return newFiles
 	}
 
-	appendedIndexes := make([]int, 0)
+	headResults := []File{}
```
Comment (jesseduffield) [inline pkg/commands/git.go]: This is a good refactor!

**Comment by jesseduffield** [comment]
LGTM :) will merge once CI passes

**Comment by antham** [comment]
done @jesseduffield

**Pull request merged**